### PR TITLE
Add legacy to native migration guides

### PR DESCRIPTION
**Pull request by Joelg96**
## Description

* Added migration guides for migrating from legacy to native file systems.
* Azure, S3, and Google

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 1]
```diff
@@ -192,3 +192,59 @@ CREATE TABLE hive.abfs_export.orders_abfs
 WITH (format = 'ORC')
 AS SELECT * FROM tpch.sf1.orders;
 ```
+
+(legacy-azure-migration)=
+## Migration to native Azure implementation
+
+Trino includes a {doc}`native Azure implementation
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-azure.md:199]: Use markdown syntax for all links so in this example:

```
[native Azure implementation](/object-storage/file-system-azure)
```

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 2]
```diff
@@ -192,3 +192,59 @@ CREATE TABLE hive.abfs_export.orders_abfs
...
+</object-storage/file-system-azure>` that does not rely on the legacy Hadoop
+implementation. Trino recommends upgrading existing deployments to the new
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-azure.md:201]: Maybe reword to create more urgency ... strong advised to switch and use the native integration

also mention that legacy will be deprecated and removed

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 3]
```diff
@@ -192,3 +192,59 @@ CREATE TABLE hive.abfs_export.orders_abfs
...
+native implementation at your earliest convenience.
+
+The following considerations apply to legacy configurations moving to the
+native implementation:
+
+* Legacy storage types (WASB and ADL) are not supported.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-azure.md:207]: This is probably not worth mentioning since it isnt supported by MS and by the legacy Hadoop file system stuff either.

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 4]
```diff
@@ -192,3 +192,59 @@ CREATE TABLE hive.abfs_export.orders_abfs
...
+* Azure Government is not supported.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-azure.md:208]: Thats no longer true with the new PR from David https://github.com/trinodb/trino/pull/23071 which will probably merge before this PR

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 5]
```diff
@@ -192,3 +192,59 @@ CREATE TABLE hive.abfs_export.orders_abfs
...
+
+To migrate a catalog to use the native filesystem implementation for Azure, make
+the following edits to your catalog configuration:
+
+1. Add the `fs.native-azure.enabled=true` catalog configuration property.
+2. Add the `azure.auth-type` catalog configuration property. Set to `ACCESS_KEY`
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-azure.md:214]: Just say "Configure the `azure.auth-type` catalog configuration property." 

Dont worry about the details .. its documented on the same page above. If you really want you can link to it

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 6]
```diff
@@ -192,3 +192,59 @@ CREATE TABLE hive.abfs_export.orders_abfs
...
+  :header-rows: 1
+   * - Legacy property
+     - Native property
+     - Notes
+   * - `hive.azure.abfs-access-key`
+     - `azure.access-key`
+     -
+   * - `hive.azure.abfs.oauth.endpoint`
+     - `azure.oauth.endpoint`
+     - Also see `azure.oauth.tenant-id` in {ref}`azure-oauth-authentication`.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-azure.md:234]: use markdown link

```suggestion
     - Also see `azure.oauth.tenant-id` in [](azure-oauth-authentication).
```

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 7]
```diff
@@ -192,3 +192,59 @@ CREATE TABLE hive.abfs_export.orders_abfs
...
+   * - `hive.azure.abfs.oauth2.passthrough`
+     - `azure.use-oauth-passthrough-token`
+     -
+  :::
+
+4. Remove the following legacy configuration properties if they exist in your
+   catalog configuration:
+
+      * `hive.azure.abfs-storage-account`
+      * `hive.azure.wasb*`
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-azure.md:250]: Reword to a text like "Add properties starting with `hive.azure.wasb`" or list them

File: docs/src/main/sphinx/object-storage/legacy-gcs.md [hunk 8]
```diff
@@ -77,3 +77,34 @@ folder defined in the JSON key file.
 Your table is now ready to populate with data using `INSERT` statements.
 Alternatively, you can use `CREATE TABLE AS` statements to create and
 populate the table in a single statement.
+
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-gcs.md:80]: Do same adjustments as in azure

File: docs/src/main/sphinx/object-storage/legacy-s3.md [hunk 9]
```diff
@@ -326,3 +326,138 @@ classpath and must be able to communicate with your custom key management system
 the `org.apache.hadoop.conf.Configurable` interface from the Hadoop Java API, then the Hadoop configuration
 is passed in after the object instance is created, and before it is asked to provision or retrieve any
 encryption keys.
+
+(legacy-s3-migration)=
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-s3.md:330]: Do same adjustments as in azure

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 10]
```diff
@@ -192,3 +192,59 @@ CREATE TABLE hive.abfs_export.orders_abfs
 WITH (format = 'ORC')
 AS SELECT * FROM tpch.sf1.orders;
 ```
+
+(legacy-azure-migration)=
```
[the remark this answers is not in the document]
Comment (jhlodin) [reply docs/src/main/sphinx/object-storage/legacy-azure.md:196]: That makes sense, sure.

File: docs/src/main/sphinx/object-storage/legacy-azure.md (hunk 5 above)
Comment (jhlodin) [reply docs/src/main/sphinx/object-storage/legacy-azure.md:214]: I disagree. Migration guides should have complete info and not require clicking around if at all possible.

File: docs/src/main/sphinx/object-storage/legacy-azure.md (hunk 7 above)
Comment (jhlodin) [reply docs/src/main/sphinx/object-storage/legacy-azure.md:250]: Suggest:
* All properties beginning with `hive.azure.wasb`

File: docs/src/main/sphinx/object-storage/legacy-azure.md (hunk 5 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/legacy-azure.md:214]: The problem is that this will get super complex with S3 and potentially any others like minio usage. We would need to at least cover required properties I think. Also .. it has to be maintained as duplicate info from now on until we remove the duplication (or the whole migration section) .. not sure if this is worth the effort.

File: docs/src/main/sphinx/object-storage/legacy-azure.md (hunk 7 above)
Comment (Joelg96) [reply docs/src/main/sphinx/object-storage/legacy-azure.md:250]: I'll just list them, I only found two

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 11]
```diff
@@ -192,3 +192,52 @@ CREATE TABLE hive.abfs_export.orders_abfs
 WITH (format = 'ORC')
 AS SELECT * FROM tpch.sf1.orders;
 ```
+
+(fs-legacy-azure-migration)=
+## Migration to native Azure implementation
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-azure.md:197]: Term "native Azure implementation" is wrong .. we need to reword here and below. Same for the other .. maybe just use the title of the doc

Azure Storage file system support

https://trino.io/docs/current/object-storage/file-system-azure.html

.. and expand to something along the lines of the content of that doc:

Trino includes a native implementation to access [Azure Storage](https://learn.microsoft.com/en-us/azure/storage/) with a catalog using the Delta Lake, Hive, Hudi, or Iceberg connectors.

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 12]
```diff
@@ -192,3 +192,52 @@ CREATE TABLE hive.abfs_export.orders_abfs
...
+
+Trino includes a [native Azure
+implementation](/object-storage/file-system-azure) that does not rely on the
+legacy Hadoop implementation. Trino strongly advises upgrading existing
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-azure.md:201]: legacy Hadoop implementation .. is also wrong.. maybe "legacy support, that is facilitates libraries from the Hadoop, Hive, and HDFS ecosystem" or something like that

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 13]
```diff
@@ -192,3 +192,52 @@ CREATE TABLE hive.abfs_export.orders_abfs
...
+deployments to the new native implementation. Legacy support will soon be
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-azure.md:202]:
```suggestion
deployments to the new native implementation. Legacy support will be
```

same in other docs

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 14]
```diff
@@ -192,3 +192,52 @@ CREATE TABLE hive.abfs_export.orders_abfs
...
+deprecated and removed.
+
+To migrate a catalog to use the native filesystem implementation for Azure, make
+the following edits to your catalog configuration:
+
+1. Add the `fs.native-azure.enabled=true` catalog configuration property.
+2. Configure the `azure.auth-type` catalog configuration property.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-azure.md:209]: link to the properties for the new support either here or after the list with a explicit sentence that says something like .. for more info on the new stuff look at x

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/object-storage/legacy-azure.md [hunk 15]
```diff
@@ -192,3 +192,55 @@ CREATE TABLE hive.abfs_export.orders_abfs
...
+  :::
+
+4. Remove the following legacy configuration properties if they exist in your
+   catalog configuration:
+
+      * `hive.azure.abfs-storage-account`
+      * `hive.azure.wasb-access-key`
+      * `hive.azure.wasb-storage-account`
+
+For more information, see the [native file system support
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-azure.md:245]: Just use this:

```
see [](/object-storage/file-system-azure).
```

also for the others

File: docs/src/main/sphinx/object-storage/legacy-s3.md [hunk 16]
```diff
@@ -326,3 +326,141 @@ classpath and must be able to communicate with your custom key management system
...
+     -
+   * - `hive.s3.endpoint`
+     - `s3.endpoint`
+     -
+   * - `hive.s3.region`
+     - `s3.region`
+     -
+   * - `hive.s3.sse.enabled`
+     - None
+     - `s3.sse.type` being set to the default value of `NONE` is equivalent to
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/legacy-s3.md:382]:
```suggestion
     - `s3.sse.type` set to the default value of `NONE` is equivalent to
```

**Review by mosabua — CHANGES_REQUESTED** [review]
Nits.. then good to go.

**Review by mosabua — APPROVED** [review]
